flow/cmd/handler.go

```diff
@@ -327,6 +328,17 @@ func (h *FlowRequestHandler) FlowStateChange(
 ) (*protos.FlowStateChangeResponse, error) {
 	logs := slog.String("flowJobName", req.FlowJobName)
 	slog.Info("FlowStateChange called", logs, slog.Any("req", req))
+	underMaintenance, err := peerdbenv.PeerDBMaintenanceModeEnabled(ctx, nil)
```

Comment:
do we want to prohibit edit peer too?

Reply:
I think edit peer should be fine, it does not lead to nondeterminism iirc